pkg/generate/app/cmd/newapp.go

```diff
@@ -243,6 +243,34 @@ func (c *AppConfig) AddArguments(args []string) []string {
 	return unknown
 }
 
+// validateBuilders confirms that all images associated with components that are to be built,
+// are builders (or we're using a docker strategy).
+func (c *AppConfig) validateBuilders(components app.ComponentReferences) error {
+	errs := []error{}
+	for _, ref := range components {
+		input := ref.Input()
+
+		// if the strategy is explicitly Docker, all repos should assume docker
+		if c.Strategy == "docker" && input.Uses != nil {
+			input.Uses.BuildWithDocker()
```

Comment:
yeah, this code had rotted away in a forgotten branch for me, so it turns out that logic was redundant now.  will clean it up, thanks for catching it.